lib/proxy-agent.js

```diff
@@ -82,40 +82,38 @@ class ProxyAgent extends DispatcherBase {
     this[kClient] = clientFactory(resolvedUrl, { connect })
     this[kAgent] = new Agent({
       ...opts,
-      connect: async (opts, callback) => {
+      connect: (opts, callback) => {
```

Comment:
I think this is unnecessary change

Reply:
tbh. This is for discussing. I agree this is reducing the maintainability. I am unsure if we save one Promise by this.